File: bin/Data/AUI/Periodic/Sources/code_aseparator.cpp
// ASeparator application source code
#include "PeriodicApp.h"

void PeriodicApp::setup_aseparator( AWidget *layout )
{
    PODVector<AWidget*> dest;
    layout->SearchWidgetClass( "TBButton", dest );
    for (unsigned ii = 0; ii < dest.Size(); ii++)
        SubscribeToEvent(dest[ii], E_WIDGETEVENT, URHO3D_HANDLER(PeriodicApp, HandleAseparatorEvent ));
}

void PeriodicApp::HandleAseparatorEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace WidgetEvent;
    AWidget* widget = static_cast<AWidget*>(eventData[P_TARGET].GetPtr());
    if ( widget == NULL ) return;
    if (eventData[P_TYPE] == UI_EVENT_TYPE_CLICK)
    {
        if (widget->GetId() == "uiseparatorcode" )
        {
            AppLog( "ASeparator support : " + widget->GetId() + " was pressed " );
            ViewCode ( "Sources/code_aseparator.cpp", widget->GetParent() );
        }
        if (widget->GetId() ==  "uiseparatorlayout" )
        {
            AppLog( "ASeparator support : " + widget->GetId() + " was pressed ");
            ViewCode ( "Scenes/layout_aseparator.ui.txt", widget->GetParent() );
        }
    }
}

File: bin/Data/AUI/Periodic/LuaScripts/code_aseparator.lua
-- ASeparator application source code


function setup_uiseparator( layout )
    local dest = layout:SearchAWidgetClass( "TBButton" )
    for i, awid in ipairs(dest) do
        SubscribeToEvent(awid, "WidgetEvent", "HandleUiseparatorEvent")
	end
end

function HandleUiseparatorEvent(eventType, eventData)
    local widget = eventData["Target"]:GetPtr("AWidget")
    if widget == nil then
 		return
	end
    if eventData["Type"]:GetInt() == UI_EVENT_TYPE_CLICK then
        if widget:GetId() == "uiseparatorcode" then
            AppLog( "ASeparator support : " .. widget:GetId() .. " was pressed " )
            ViewCode ( "LuaScripts/code_aseparator.lua", widget:GetParent())
        end
        if widget:GetId() ==  "uiseparatorlayout" then
            AppLog( "ASeparator support : " .. widget:GetId() .. " was pressed ")
            ViewCode ( "Scenes/layout_aseparator.ui.txt", widget:GetParent())
        end
    end
end

